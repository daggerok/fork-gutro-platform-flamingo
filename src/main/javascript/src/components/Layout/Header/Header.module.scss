
.header {
  height: 70px !important;
  display: flex;
  flex-wrap: wrap;
  align-items: center;

  .title {
    font-family: 'Sedgwick Ave', cursive;
    color: white;
    font-size: 24px;
    margin: 0 20px 0 10px;
  }
  .logo {
    display: inline-block;
    height: 42px;
  }
  .flagContainer {
    margin-left: 10px;
  }
}
