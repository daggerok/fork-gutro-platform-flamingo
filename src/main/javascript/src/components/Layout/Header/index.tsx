import React from 'react';
import { Layout } from 'antd';
const { Header: AntdHeader } = Layout;
import EnvironmentFlag from '~/components/Common/EnvironmentFlag';
import { HeaderProps } from '../types';
import styles from './Header.module.scss';
import logo from './logo.png';

const Header: React.FC<HeaderProps> = ({ children }: HeaderProps) => (
  <AntdHeader 
    data-testid="header"
    className={styles.header}
  >
    <img
      src={logo}
      data-testid="header-logo"
      className={styles.logo}
    />
    <span 
      data-testid="header-title"
      className={styles.title}
    >
      Flamingo
    </span>
    <div 
      data-testid="header-flag-container"
      className={styles.flagContainer}
    >
      <EnvironmentFlag />
    </div>
    { children }
  </AntdHeader>
);

export default Header;
