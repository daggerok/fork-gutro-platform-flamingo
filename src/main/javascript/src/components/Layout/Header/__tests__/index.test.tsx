import React from 'react';
import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Header from '..';

jest.mock('./../../../Common/EnvironmentFlag', () => 'environment-flag');

describe('<Header />', () => {

  it('matches snapshot', () => {
    act(() => {
      render(<Header children />);
    });

    const component = screen.getByTestId('header');

    expect(component).toMatchSnapshot();
  });

  it('uses correct logo image src', () => {
    act(() => {
      render(<Header children />);
    });

    const logo = screen.getByTestId('header-logo');

    expect(logo.getAttribute('src')).toEqual('logo.png');
  });

  it('renders title', () => {
    act(() => {
      render(<Header children />);
    });
    
    const title = screen.getByTestId('header-title');

    expect(title.innerHTML).toEqual('Flamingo');
  });

});