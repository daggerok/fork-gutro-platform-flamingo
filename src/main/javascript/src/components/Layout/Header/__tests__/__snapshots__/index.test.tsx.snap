// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<header
  class="ant-layout-header header"
  data-testid="header"
>
  <img
    class="logo"
    data-testid="header-logo"
    src="logo.png"
  />
  <span
    class="title"
    data-testid="header-title"
  >
    Flamingo
  </span>
  <div
    class="flagContainer"
    data-testid="header-flag-container"
  >
    <environment-flag />
  </div>
</header>
`;
